File: best-time-to-buy-and-sell-stock/best_time_to_buy_and_sell_stock.go
func maxProfit(prices []int) int {
	if len(prices) <= 1 {
		return 0
	}
	minimum := prices[0]
	profit := 0
	for i := 1; i < len(prices); i++ {
		if prices[i] > prices[i-1] {
			minimum = min(minimum, prices[i-1])
			profit = max(profit, prices[i]-minimum)
		}
	}
	return profit
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: best-time-to-buy-and-sell-stock/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1:
            return 0

        minimum = prices[0]
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                minimum = min(minimum, prices[i-1])
                profit = max(profit, prices[i]-minimum)
        return profit
